homeassistant/components/icloud/config_flow.py

```diff
@@ -173,22 +178,23 @@ async def async_step_user(self, user_input=None):
 
         return await self._validate_and_create_entry(user_input, "user")
 
-    async def async_step_reauth(self, user_input=None):
-        """Update password for a config entry that can't authenticate."""
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Initialise re-authentication."""
         # Store existing entry data so it can be used later and set unique ID
         # so existing config entry can be updated
-        if not self._existing_entry:
-            await self.async_set_unique_id(user_input.pop("unique_id"))
-            self._existing_entry = user_input.copy()
-            self._description_placeholders = {"username": user_input[CONF_USERNAME]}
-            user_input = None
-
+        await self.async_set_unique_id(self.context["unique_id"])
+        self._existing_entry = {**entry_data}
```

Comment:
It's confusing to call the attribute existing entry when it's the existing entry data.

Reply:
Renamed in #77582